corehq/apps/reports/views.py

```diff
@@ -1765,21 +1776,22 @@ def _error(msg):
             edit_session_data[USERCASE_ID] = usercase_id
 
         case_blocks = extract_case_blocks(instance, include_path=True)
-        if form.form_type == 'advanced_form':
-            datums = EntriesHelper(form.get_app()).get_datums_meta_for_form_generic(form)
-            for case_block in case_blocks:
+        datums = EntriesHelper(form.get_app()).get_datums_meta_for_form_generic(form)
+        for case_block in case_blocks:
+            if form.form_type == 'advanced_form':
                 path = case_block.path[0]  # all case blocks in advanced forms are nested one level deep
                 matching_datums = [datum for datum in datums if datum.action.form_element_name == path]
                 if len(matching_datums) == 1:
                     edit_session_data[matching_datums[0].datum.id] = case_block.caseblock.get(const.CASE_ATTR_ID)
-        else:
-            # a bit hacky - the app manager puts the main case directly in the form, so it won't have
-            # any other path associated with it. This allows us to differentiate from parent cases.
-            # You might think that you need to populate other session variables like parent_id, but those
-            # are never actually used in the form.
-            non_parents = filter(lambda cb: cb.path == [], case_blocks)
-            if len(non_parents) == 1:
-                edit_session_data['case_id'] = non_parents[0].caseblock.get(const.CASE_ATTR_ID)
+            else:
+                if case_block.path == [] and len(get_non_parents(case_blocks)) == 1:
+                    # a bit hacky - the app manager puts the main case directly in the form, so it won't have
+                    # any other path associated with it. This allows us to differentiate from parent cases.
+                    edit_session_data['case_id'] = case_block.caseblock.get(const.CASE_ATTR_ID)
+                elif is_subcase(case_block) and not case_block.caseblock.get(const.CASE_ATTR_ID):
```

Comment:
session data DOES get a value set for the subcase ID if the subcase exists. I still need to find where this happens. It may be that wherever that is might be a better place to do all this. But considering it happens here for advanced forms, here seemed to be the right place to do it for basic forms too.